optimum/exporters/onnx/model_configs.py

```diff
@@ -634,10 +608,8 @@ def __init__(
     def inputs(self) -> Dict[str, Dict[int, str]]:
         common_inputs = super().inputs
 
-        if not self.legacy and not self._config.alibi and self.task in ["text-generation", "feature-extraction"]:
-            # When alibi is used, position_ids are not used in Falcon.
-            # Reference: https://github.com/huggingface/transformers/blob/v4.34.0/src/transformers/models/falcon/modeling_falcon.py#L1116
-            common_inputs["position_ids"] = {0: "batch_size", 1: "sequence_length"}
+        if self._config.alibi:
+            common_inputs.pop("position_ids", None)
```

Comment:
not very important but here why can we remove the check for legacy models ? (I don't think legacy models are tested very well if not at all, also and we should definitely remove their support in v2 so not very important in all cases)

Reply:
legacy is tested in https://github.com/huggingface/optimum/blob/main/tests/onnxruntime/test_decoder.py#L635, we compare the legacy+non-merged with merged

Reply:
ok very clear, thanks @IlyasMoutawwakil !!